Add golf vehicle launch

NilaySener opened the pull request:
Created launch and description file for golf vehicle.

NilaySener committed:
Add golf vehicle launch

MertClk commented:
@ismetatabay could you please review and test this PR?

ismetatabay commented on golf_vehicle_description/config/simulator_model.param.yaml:
```diff
@@ -0,0 +1,18 @@
+    steer_time_delay: 0.24
+    steer_time_constant: 0.27
+    x_stddev: 0.0001 # x standard deviation for dummy covariance in map coordinate
+    y_stddev: 0.0001 # y standard deviation for dummy covariance in map coordinate
```
please add new line to this file, there is no way to tell the difference between a file where there is a newline at the end and one where is not. (Diff)

ismetatabay commented on golf_vehicle_description/config/vehicle_info.param.yaml:
```diff
@@ -0,0 +1,12 @@
+    left_overhang: 0.14  # between left wheel center and vehicle left
+    right_overhang: 0.14  # between right wheel center and vehicle right
+    vehicle_height: 1.86 
+    max_steer_angle: 0.6105 # [rad] check again!(temp)
```
Same

ismetatabay commented on golf_vehicle_launch/launch/vehicle_interface.launch.xml:
```diff
@@ -1,4 +1,4 @@
 <launch>
   <arg name="vehicle_id" default="$(env VEHICLE_ID default)"/>
-</launch>
+</launch>
```
Same

ismetatabay reviewed (commented):
@NilaySener Thanks for your PR,

I confirmed this PR work well, but golf model exceeds slightly ground to underground.

## Dimensions (seems to me ok)
Sample Vehicle (Lexus Mirrors Excluded):
![Selection_036](https://user-images.githubusercontent.com/56237550/210057370-a3c949fa-095a-4552-a320-68c84ee8c2de.png)

Our Golf (Mirrors Included):
![Selection_037](https://user-images.githubusercontent.com/56237550/210057464-3ef60bf3-cca3-4b6c-9d4f-81d6aec09a49.png)

## Orbit View (Golf exceeds ground plane)

Sample Vehicle (Lexus):

![Selection_039](https://user-images.githubusercontent.com/56237550/210058415-fcb37d09-5fdc-4677-b68c-cc40d2be4639.png)

Golf: 
![Selection_040](https://user-images.githubusercontent.com/56237550/210058440-3184afef-e86d-4094-9f64-8f41f8a95758.png)

![Selection_041](https://user-images.githubusercontent.com/56237550/210058447-0ac8fc5a-96be-4417-b3e4-8208f1b4a932.png)

NilaySener committed:
update z-axis of the golf_vehicle

NilaySener commented:
Hi @ismetatabay , thanks for the review. I changed the z-axis to bring the vehicle to the map level and added the new lines you mentioned. Can you check if the coordinates are correct?

NilaySener committed:
update z-axis of the golf_vehicle

ismetatabay commented on golf_vehicle_description/urdf/vehicle.xacro:
```diff
@@ -0,0 +1,21 @@
+  <link name="base_link">
+    <visual>
+      <!--<origin xyz="${vehicle_info['/**']['ros__parameters']['wheel_base']/2.0} 0 0" rpy="${pi/2.0} 0 ${pi/2.0}"/>-->
+      <origin xyz="1.3772 0 -0.1" rpy="${pi/2.0} 0 -${pi/2.0}"/>
```
There is still golf exceeds ground slightly,I think this will solve our problem:
```suggestion
      <origin xyz="1.3772 0 0" rpy="${pi/2.0} 0 -${pi/2.0}"/>
```

NilaySener replied:
I have committed your suggestion. can you check the vehicle again?

NilaySener committed:
Update golf_vehicle_description/urdf/vehicle.xacro

Co-authored-by: Ismet Atabay <56237550+ismetatabay@users.noreply.github.com>

ismetatabay reviewed (approved):
LGTM

![Selection_043](https://user-images.githubusercontent.com/56237550/211536599-888eeff2-4178-4d0f-9154-1b831a986507.png)
